### PR TITLE
search in right context for parent context vclusters

### LINKED ISSUE #786: Nested vcluster commands expect host cluster context name
### What happened?

After creating a vcluster inside another vcluster the `vcluster` commands expect the new kubeconfig context to be named the same as the original context.

```
$> vcluster list                           
fatal  find vcluster: load kube config: context "tge" does not exist
```

### What did you expect to happen?

It should have listed the vclusters (and not found any) in the new vcluster environment.

### How can we reproduce it (as minimally and precisely as possible)?

```
$> vcluster list

 NAME    NAMESPACE   STATUS    CONNECTED   CREATED                         AGE    
 touge   vcluster    Running               2022-10-13 15:28:47 -0700 PDT   27m8s  
```

Then connect to the vcluster

```
$> vcluster connect --kube-config ./vc-touge.yaml --update-current=false touge
done √ Virtual cluster kube config written to: ./vc-touge.yaml
- Use `vcluster connect touge -n vcluster -- kubectl get ns` to execute a command directly within this terminal
- Use `kubectl --kubeconfig ./vc-touge.yaml get namespaces` to access the vcluster
Forwarding from 127.0.0.1:10000 -> 8443
Forwarding from [::1]:10000 -> 8443
```

And in another window

```
$> export KUBECONFIG="$(pwd)/vc-touge.yaml"

$> vcluster list                           
fatal  find vcluster: load kube config: context "tge" does not exist
```

If you edit the `vc-touge.yaml` and change the context to be named `tge` then it works as expected.

### Anything else we need to know?

_No response_

### Host cluster Kubernetes version

<details>

```console
$> kubectl version
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.10", GitCommit:"eae22ba6238096f5dec1ceb62766e97783f0ba2f", GitTreeState:"clean", BuildDate:"2022-05-24T12:56:35Z", GoVersion:"go1.16.15", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"23+", GitVersion:"v1.23.10-eks-15b7512", GitCommit:"cd6399691d9b1fed9ec20c9c5e82f5993c3f42cb", GitTreeState:"clean", BuildDate:"2022-08-31T19:17:01Z", GoVersion:"go1.17.13", Compiler:"gc", Platform:"linux/amd64"}
```

</details>

### Host cluster Kubernetes distribution

<details>

```
EKS v1.23.10
```

</details>

### vlcuster version

<details>

```console
$> vcluster --version
vcluster version 0.12.2
```

</details>

### Vcluster Kubernetes distribution(k3s(default)), k8s, k0s)

<details>

```
k3s
```

</details>

### OS and Arch

<details>

```
OS: MacOS
Arch: amd64
```
</details>

### PR DESCRIPTION
**Pull request by bagulm123**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind bugfix

**What does this pull request do? Which issues does it resolve?** (use `resolves #<issue_number>` if possible) 
resolves #786 

**Please provide a short message that should be published in the vcluster release notes**
Search in right context for parent context vclusters

**What else do we need to know?**

### CODE REVIEW

File: cmd/vclusterctl/cmd/find/find.go [hunk 1]
```diff
@@ -84,11 +85,22 @@ func ListVClusters(context, name, namespace string) ([]VCluster, error) {
 	}
 
 	if vClusterName != "" {
+		currentKubeConfig := os.Getenv("KUBECONFIG")
+		// check in all contexts - assuming parent context will be one of in all contexts.
```
[the remark this answers is not in the document]
Comment (matskiv) [reply cmd/vclusterctl/cmd/find/find.go:89]: This is a good point, the parent context might be present in a kubeconfig that was referenced by the env var too, and we won't have access to it.

**Comment by mtougeron** [comment]
It's also worth noticing what I discovered this morning in https://github.com/loft-sh/vcluster/issues/786#issuecomment-1279011728

**Comment by matskiv** [comment]
Actually, my previous comment would solve a potential issue, but not the one reported by @mtougeron in #786 
Currently `vcluster list` will show vclusters from the parent cluster if it detects that the current context is vcluster context, but in #786 it is requested to list only vclusters from the current context without trying to read the parent context.

**Comment by rohantmp** [comment]
@matskiv @mahendrabagul 

I don't think it's a safe assumption that the config will have access to the parent cluster, and we should respect `--kubeconfig` and `KUBECONFIG` and I think we shouldn't try to access other kubeconfigs.

 IMO: Since we support nesting, `vcluster list` should not return sibling clusters, but we could warn that we are listing while connected to a vcluster.

**Review by rohantmp — CHANGES_REQUESTED** [review]
As per previous comment, I think we should review behavior around listing sibling clusters by looking up parent config.
We could do a quick fix that preserves the current behaviour by failing gracefully if parent context is not available.

**Comment by matskiv** [comment]
> I don't think it's a safe assumption that the config will have access to the parent cluster, and we should respect --kubeconfig and KUBECONFIG and I think we shouldn't try to access other kubeconfig files. 

@rohantmp yes, I agree. On Fri I was so focused on how to make it work with other kubeconfig files, that I didn't think whether it is a good idea. :slightly_smiling_face: 

> IMO: Since we support nesting, vcluster list should not return sibling clusters, but we could warn that we are listing while connected to a vcluster.

Current `vcluster list` and `vcluster connect` behavior returns/connects to sibling clusters. I would keep it as is, especially considering that nested clusters are arguably a less popular use case. But I think we can support both use cases. I describe it in the last paragraph [of this comment](https://github.com/loft-sh/vcluster/issues/786#issuecomment-1279210679).

**Review by rohantmp — APPROVED** [review]
LGMT :)

**Pull request merged**